### PR TITLE
Use mock module and move tests

### LINKED ISSUE #45: Move mock and test out of top level
Diffcalc exposes both `mock` and `test` at the top level, both of which could cause issues when trying to use diffcalc within a large codebase.

In particular this has caused an issue in GDA where doing `import mock` from other Jython files uses the `diffcalc` version rather than the `mock` package we have installed in `site-packages`.

For the `mock` package can we just list it as a dependency in the `setup.py` and remove it from the codebase entirely?

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes #45 

To test:
* Confirm tests still pass despite `mock` and `test` being moved out of top level

### CODE REVIEW

**Comment by joeshannon** [comment]
Ideally the tests would be separate from the main module(s) so that they wouldn't be present in the application.
Something like a new `src` directory containing the diffcalc module (and the other modules) and then leaving the tests where they are?
It looks like it would be more work as would have to check the structure still works and might have other implications for development (installing package in dev mode etc.)
Also the GDA diffcalc ScriptProject configuration would be updated.

Due to these complications maybe it is easier to move the tests as you have done but I might ask for a second opinion.

I think it might also be good to wait until we can get the CI running again (#46) to verify the tests can still all run as expected.

**Comment by joeshannon** [comment]
We should merge #46  first, then fix-up this so that the tests pass.

**Comment by DominicOram** [comment]
@joeshannon and @isikhar, this ok?

File: .github/workflows/main.yml
Comment (joeshannon) [inline .github/workflows/main.yml:25]: Could specify the exact version here as done for the other packages but not a major issue (I doubt they are going to release a python 2 package version which breaks behaviour).

**Review by joeshannon — APPROVED** [review]
Looks good to me. I think Irakli was previously happy with these changes.

**Comment by isikhar** [comment]
These changes are fine with me as well.

File: .github/workflows/main.yml
Comment (DominicOram) [reply .github/workflows/main.yml]: Pinned to 3.0.5

**Comment by DominicOram** [comment]
Is the CI known to be broken?

**Comment by joeshannon** [comment]
No I don't think so. Not sure why it's not running. I see that it run for your fork. 

I'm not hugely familiar with GH actions, we currently have
```
name: Diffcalc build

on:
  push:
  pull_request:
```
Does that seem correct?

**Comment by DominicOram** [comment]
So it did (see https://github.com/DominicOram/diffcalc/actions/runs/3338486618/jobs/5526204223). Yes, those actions seem correct to me. Are we happy to chalk it up to GH weirdness and get I'll merge this?

**Pull request merged**
